awx/main/management/commands/provision_instance.py

```diff
@@ -37,7 +41,10 @@ def _register_hostname(self, hostname):
     def handle(self, **options):
         if not options.get('hostname'):
             raise CommandError("Specify `--hostname` to use this command.")
-        self.uuid = settings.SYSTEM_UUID
+        if options['is_isolated']:
```

Comment:
I was hoping to avoid adding a new flag to the management command and dependent tooling, but I believe it is necessary.  

`instance.is_isolated()`, etc. require the instance groups to be registered to work properly. Since instance _groups_ are registered after _instances_, we can't tell if an instance is isolated during install time when this command is used.